doc/tasks/readme.md

```diff
@@ -0,0 +1,737 @@
+sind. Für obiges Beispiel sind folgende Komponenten erforderlich:
+
+- InventoryComponent im Spielcharakter (zur Realisierung des Spielerinventars)
+- CollisionComponent im Spielcharakter und im Item, um ein Event auszulösen, sobald der
+  Spielcharakter über das Item läuft
+- alternativ ein InteractionComponent im Item, falls das Item per dedizierter
+  Spielendenaktion (bspw. Tastendruck) aufgehoben werden soll
+
+### Spielszenario
+
+Ein Spielszenario beschreibt, welche Spielmechaniken, welche Entitätstypen und auf welche
+Art diese in einem konkreten Dungeonlevel in Verbindung gebracht werden, um eine abstrakte
+Aufgabe zu realisieren. Ein Spielszenario beschreibt nicht, welche visuellen
+
+Beispiel: In einem Raum befindet sich ein NPC und verschiedene Schalter. Der NPC stellt dem
+Spieler eine Frage und sagt ihm, welcher Hebel für welche Antwort steht.
+
+### Steuermechanismen
+
+Steuermechanismen sind alle Mechanismen des Systems, die den Ablauf der Umsetzung einer
+Aufgabe in ein Spielszenario beeinflussen.
+
+Steuermechanismen grenzen sich von Spielmechaniken ab, indem sie die übergeordneten
+Mechanismen realisieren, die sich nicht ausschließlich auf Game-Design Elemente beziehen.
+Hierzu zählen unter anderem das Logging von Antworten, die Organisation der Aufgabenfolge
+und Aufgabenverschachtelung.
+
+## Aufgabentypen
+
+### Single-Choice
+
+Die Aufgabenstellung besteht aus einem Fragetext und einer Menge an vorgegebenen
+Antwortmöglickeiten. Nur eine Antwort ist richtig. Die richtige Antwort muss ausgewählt
+werden.
+
+### Multiple-Choice
+
+Die Aufgabenstellung besteht aus einem Fragetext und einer Menge an vorgegebenen
+Antwortmöglickeiten. Mehrere Antworten sind richtig. Es müssen alle richtigen Antworten
+markiert werden. Es dürfen keine falschen Antworten markiert werden. Die Reihenfolge spielt
+keine Rolle.
+
+### Lückentext
+
+Die Aufgabenstellung besteht aus einem Fragetext. Die Antwort besteht aus einem oder
+mehreren Freitexten. Es muss jeweils die richtige Antwort in den Freitext-Stellen eingegeben
+werden. Die richtige Antwort kann ein konkreter Wert oder eine Eingabe passend zu einem
+regulären Ausdruck sein.
+
+### Ersetzen
+
+Aus einer Gesamtmenge aus $l$ Elementen (mit $0 < l$) muss eine Menge mit $m$ Elementen (mit $0 < n$) durch eine
+zweite Menge mit $m$ Elementen (mit $0 < m$) ersetzt werden.
+Hierdurch wird die Gesamtmenge der Elemente sowie die Anzahl der enthaltenen Elemente verändert.
+
+### Zuordnen
+
+Zuordnen bezeichnet im Allgemeinen die Aufgabe, Elemente aus einer Menge $A$ (z.B.
+Antwortmöglichkeiten) die Elemente aus einer anderen Menge $B$ zuzuordnen (bspw. Array-Index,
+Reihe-Spalte Kombination einer Matrix). Hierbei muss nicht jedes Element aus $A$ der
+Antwortmöglichkeiten auch einem Element aus $B$ zugeordnet werden. Allerdings muss jedem Element aus $B$ ein
+Element aus $A$ zugeordnet werden. Die gesamte Menge $A$ der Antwortmöglichkeiten muss den Studierenden
```

Comment:
Ist das nicht einfach $f: B \to A$? Eine Abbildung aus B auf A?

1. Ich muss nicht jedem Element aus $A$ etwas zuordnen.
2. Ich muss nicht jedes Element aus $B$ zuordnen.
3. Ich kann einem Element aus $A$ mehrere Elemente aus $B$ zuordnen.
4. Ich kann ein Element aus $B$ mehreren Elementen aus $A$ zuordnen (???!) => aber dann verlassen wir den klassischen Begriff der "Abbildung".

Reply:
Das als Abbildung zu definieren ist schwierig, da die Zuordnung der Elemente zueinander ja durch diskrete Entitäten passiert. Das heißt, ich kann nicht einfach so ein Element aus $A$ mehreren Elemente aus $B$ zuordnen (wie das ja bei einer Abbildung möglich wäre), weil ich ja nur einmal dieses Element aus $A$ als Entität habe (so zumindest die initiale Idee).

Reply:
Wenn ich eine Abbildung habe, dann würde ich $b \in B$ auf $a \in A$ zuordnen. Dabei könnte ich aber im Grunde nur (1)-(3), also jedem $b$ genau ein $a$ zuordnen. Das würde doch genau zu Deiner Aussage passen, oder? (4) ist eh Käse, weil in Abbildungen nicht erlaubt.

Reply:
Dass jedem $b$ genau ein $a$ zugeordnet werden kann, stimmt. Das Problem ist aber (3), was diese Restriktion in die andere Richtung, also auch jedem $a$ nur genau ein $b$ zuzuordnen, ja aufhebt. Die brauchen wir aber, wenn wir sagen $A$ und $B$ bestehen aus diskreten Entitäten, die wir nicht einfach duplizieren können.
Wenn in $A$ 4 verschiedene Schriftrollen sind und $B$ 3 verschiedene Briefkästen, dann kann ich keine der Schriftrollen $a \in A$ in mehrere Briefkästen werfen (was ja (3) entspräche), sondern nur in genau einen.

Reply:
@malt-r OK, Du hast die Abbildung umgedreht, also nicht $f: B \to A$ (wie ich angenommen hatte), sondern Du meinst $f: A \to B$. $A$ wären im Beispiel Schriftrollen, $B$ Briefkästen. Ich muss eine Abbildung aus $A$ in $B$ definieren, d.h. ich muss $a \in A$ auf $b \in B$ abbilden. (1) bis (3) passen also (wobei (3) dann natürlich umgedreht ist: also einem $a$ kann maximal ein $b$ zugeordnet werden, aber ich kann natürlich einem $b$ mehrere $a$ zuordnen - ich kann ja mehrere Schriftrollen in den selben Briefkasten werfen - die Abbildung ist also nicht injektiv und nicht surjektiv), und (4) ist eh Quatsch, weil mathematisch nicht erlaubt.

Wenn Du da zustimmst, würde ich den Text gern entsprechend anpassen. Wir sprechen hier über eine "partielle Funktion".

Reply:
Prinzipiell passt das besser, das stimmt. Wenn ich richtig liege, fällt damit allerdings die Anforderung weg, dass auch **jedem** $b \in B$ ein $a \in A$ zugewiesen werden muss. Also, dass in jedem Briefkasten auch mindestens eine Schriftrolle liegt. Beim nochmaligen drüber nachdenken kommt mir diese Anforderung aber sowieso etwas an den Haaren herbeigezogen vor, von daher wäre ich mit der Definition als partielle Funktion happy.